### DISCUSSION TITLE
Which environment should i use to complie?

### DISCUSSION THREAD

**Discussion by hkpmatt** [mriscoc/Ender3V2S1#315] [Q&A]
So since my bl touch install in such a way
![install](https://user-images.githubusercontent.com/70888649/186946607-2f9760a7-e864-497a-b545-76c03a928008.png)
I choose to complie my own BLT configuration.

I wonder which environment should i choose from
![image](https://user-images.githubusercontent.com/70888649/186946927-5fa0b909-e30e-4e18-839b-723bb46ca809.png)

I use a Ender 3 v2, 4.2.2 board.

**Comment by ghost** [comment]
i have the same printer and the same sensor and i installed it using the same connector for all wires. I didn't make the same connection as yours and it's working perfectly, I didn't even have a problem with the compilation

**Reply by hkpmatt** [reply]
yes i know the installation suggests to install into the x o instead of the z switch port.
However, my purchase of the touch come with the port that dont fit on the x o port

**Comment by mriscoc** [comment]
Current source code doesn't support that BLTouch connection, you must to download the repository, edit the configuration files and compile your own version of the firmware.

https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch

If you have a **RET6** chip use the environment `STM32F103RE_creality`, if it is **RCT6** use `STM32F103RC_creality`.
